src/figuresClasses.ts

```diff
@@ -1,11 +1,114 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+function roundTo2FloatingPoints(num: number): number {
+  return Math.floor(num * 100) / 100;
+}
+
+function validateTriangle(a: number, b: number, c: number): void {
+  if (!a || !b || !c || a <= 0 || b <= 0 || c <= 0) {
+    throw new Error('Triangle side should be greater than 0');
+  }
+
+  const sorted: number[] = [a, b, c].sort((side1, side2) => side2 - side1);
+
+  if (sorted[0] >= sorted[1] + sorted[2]) {
+    throw new Error(
+      // eslint-disable-next-line max-len
+      'Triangle longest side should not be equal or greater the sum of the remaining sides',
+    );
+  }
+}
 
-export class Circle implements Figure {}
+export class Triangle implements Figure {
+  public readonly shape: Shape;
 
-export class Rectangle implements Figure {}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    validateTriangle(a, b, c);
+    // this.a = a;
+    // this.b = b;
+    // this.c = c;
```

Comment:
The lines for initializing `this.a`, `this.b`, and `this.c` are commented out. Ensure these properties are initialized in the constructor if they are needed elsewhere in the class.